src/Compilers/Core/Portable/MetadataReference/ModuleMetadata.cs

```diff
@@ -40,6 +45,10 @@ private ModuleMetadata(ModuleMetadata metadata)
             : base(isImageOwner: false, id: metadata.Id)
         {
             _module = metadata.Module;
+            // ensure that we keep the owner rooted so that it can't get GC'ed why we're alive.
+            _owner = metadata._owner;
+            // however, as we're not the image owner, we will never dispose the owner.  Only the single image owner can be responsible for that.
+            _disposeOwner = false;
```

Comment:
😨 This is super concerning. The original metadata could dispose the owner and this one won't be able to keep it alive.

Reply:
it's a good point.  @tmat what is the expected coordination between metadata instances, cloning, and ownership/cleanup?

@sharwell we could potentially use a RCD here right if we want to make sure this only goes away on the last metadata disposal?

Reply:
@tmat That makes sense.